File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/PricingInfoDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for course pricing information.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PricingInfoDTO {
    private Float price;
    private Float listPrice;
    private Float savingPrice;
    private String currency;
    private String priceString; // e.g., "Free"
    private String currencySymbol;
    private Integer discountPercent;
    private String discountDeadlineText; // e.g., "4 days"
    private String couponCode;
    private Integer usesRemaining;
    private Integer maximumUses;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/PricingInfoDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for course pricing information.
 */
data class PricingInfoDTO(
    val price: Float? = null,
    val listPrice: Float? = null,
    val savingPrice: Float? = null,
    val currency: String? = null,
    // e.g., "Free"
    val priceString: String? = null,
    val currencySymbol: String? = null,
    val discountPercent: Int? = null,
    // e.g., "4 days"
    val discountDeadlineText: String? = null,
    val couponCode: String? = null,
    val usesRemaining: Int? = null,
    val maximumUses: Int? = null,
)
